spec/lib/users/transfer_referent_spec.rb

```diff
@@ -0,0 +1,49 @@
+require Rails.root.join("lib/users/transfer_referent")
+
+describe Users::TransferReferent do
+  subject { described_class.new(source_referent_id: source_referent.id, target_referent_id: target_referent.id).call }
+
+  let(:user) { create(:user) }
+  let(:source_referent) { create(:agent) }
+  let(:target_referent) { create(:agent) }
+
+  before do
+    source_referent.users << user
+    stub_rdv_solidarites_assign_referent(
+      user.rdv_solidarites_user_id,
+      target_referent.rdv_solidarites_agent_id
+    )
+    stub_rdv_solidarites_assign_referent(
+      user.rdv_solidarites_user_id,
+      source_referent.rdv_solidarites_agent_id
+    )
+    allow($stdout).to receive(:puts)
```

Comment:
Je préfère qu'on ne fasse plus du tout référence à l'entrée et à la sortie standard dans les tests pour éviter de se retrouver dans la situation problématique qu'avait été résolue #2243 . 
On peut à la place utiliser (`expect_any_instance_of(..).to receive(:puts).with ...`).

D'ailleurs si on veut être picky on pourrait dire que c'est plus au niveau de la task qu'on devrait afficher les différents messages de sorties.

Reply:
ah oui c'est vrai c'était vraiment pas terrible comme problème ^^